cirq/protocols/unitary.py

```diff
@@ -90,3 +102,25 @@ def unitary(val: Any,
                         "has no _unitary_ method.".format(type(val)))
     raise TypeError("object of type '{}' does have a _unitary_ method, "
                     "but it returned NotImplemented.".format(type(val)))
+
+
+def has_unitary(val: Any) -> bool:
+    """Returns whether the value has a unitary matrix representation.
+
+    Returns:
+        If `val` has a _has_unitary_ method and its result is not
+        NotImplemented, that result is returned. Otherwise, if the value
+        has a _unitary_ method return if that has a non-default value.
+        Returns False if neither function exists.
+    """
+    getter = getattr(val, '_has_unitary_', None)
+    result = NotImplemented if getter is None else getter()
+
+    if result is not NotImplemented:
+        return result
+
+    # No _has_unitary_ function, use _unitary_ instead
+    getter = getattr(val, '_unitary_', None)
```

Comment:
Delegate to `return protocols.unitary(val, None) is not None` instead of re-implementing it.